lib/cictl/indexer.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require "socket"
+require "traject"
+
+require_relative "common"
+
+module CICTL
+  class Indexer
+    include Common
+    DEFAULT_READER_NAME = "jsonl"
+    DEFAULT_WRITER_NAME = "localhost"
+    COLLECTION_MAP_FILE = "collection_code_to_original_from.yaml"
+
+    def initialize(reader: nil, writer: nil)
+      @reader = reader
```

Comment:
I think maybe you think you're calling methods here, but there aren't any by those names. All the logic for finding readers/writers is confusing to me, and might be worth a refactor.